From 356ed8564d82be058f352a55b88721bc4123f40b Mon Sep 17 00:00:00 2001
From: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
Date: Mon, 28 Apr 2025 09:13:23 -0300
Subject: [PATCH] chrome: compose_session: Fix missing template argument

Fix the following compilation error:

```
| ../../chrome/browser/compose/compose_session.cc:553:64: error:
    alias template 'ModelExecutionSessionCallbackWithLogging' requires template
    arguments; argument deduction only allowed for class templates
|       |   optimization_guide::ModelExecutionSessionCallbackWithLogging callback =
|       |                                                                ^
| ../../components/optimization_guide/core/model_quality/
    model_execution_logging_wrappers.h:84:1: note: template is declared here
|       | using ModelExecutionSessionCallbackWithLogging =
|       | ^
| 1 error generated.
```

This patch will be dropped once clang >= 19.1.0 is available, which
includes support for class template argument deduction (CTAD) for type
alias templates (P1814R0 [0]) (#54051 [1]).

Note that currently meta-clang [2] provides the clang version `18.1.6`.

[0] https://wg21.link/p1814r0
[1] https://github.com/llvm/llvm-project/issues/54051
[2] https://github.com/kraj/meta-clang

Upstream-Status: Inappropriate [specific to older versions of clang]
Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
---
 chrome/browser/compose/compose_session.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/chrome/browser/compose/compose_session.cc b/chrome/browser/compose/compose_session.cc
index 7e34cd3457..bec8020b97 100644
--- a/chrome/browser/compose/compose_session.cc
+++ b/chrome/browser/compose/compose_session.cc
@@ -550,7 +550,7 @@ void ComposeSession::RequestWithSession(
   // execution in case request fails.
   compose::LogComposeRequestReason(request_reason);
 
-  optimization_guide::ModelExecutionSessionCallbackWithLogging callback =
+  optimization_guide::ModelExecutionSessionCallbackWithLogging<optimization_guide::proto::ComposeLoggingData> callback =
       base::BindRepeating(&ComposeSession::ModelExecutionCallback,
                           weak_ptr_factory_.GetWeakPtr(),
                           std::move(request_timer), request_id_, request_reason,
